From ee6911a53535de1dd319f496a311443f7490d562 Mon Sep 17 00:00:00 2001
From: varigigi <pierluigi.p@variscite.com>
Date: Wed, 19 Jul 2017 22:54:02 +0200
Subject: [PATCH 3/3] manage at runtime properties related to boot device

When booting from sd, all Variscite iMX6 SoMs use mmcblk1.
When booting from emmc, VAR-DART uses mmcblk2 while all the other
Variscite iMX6 SoMs use mmcblk0.
Althought this info is present in fstab, Android framework does not use it
to initialize ro.internel.storage_size and ro.frp.pst, but hardcode them
in BoardConfig.mk => in build.prop
To fix this, let's manage them at runtime if needed.
---
 init/property_service.cpp | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 5c1ae79..7616fe4 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -517,11 +517,39 @@ void load_recovery_id_prop() {
     close(fd);
 }
 
+/* When booting from sd, all Variscite iMX6 SoMs use mmcblk1.
+ * When booting from emmc, VAR-DART uses mmcblk2 while all the other
+ * Variscite iMX6 SoMs use mmcblk0.
+ * Althought this info is present in fstab, Android framework does not use it
+ * to initialize ro.internel.storage_size and ro.frp.pst, but hardcode them
+ * in BoardConfig.mk => in build.prop
+ * To fix this, let's manage them at runtime if needed.
+ */
+void variscite_system_props() {
+    std::string ro_boot_storage_type = property_get("ro.boot.storage_type");
+    std::string ro_frp_pst = property_get("ro.frp.pst");
+    std::string ro_hardware = property_get("ro.hardware");
+
+    if ((ro_boot_storage_type == "sd") || (!ro_frp_pst.empty())) {
+        /* already set */
+        return;
+    }
+
+    if (ro_hardware == "var-som-mx6") {
+        property_set("ro.internel.storage_size", "/sys/block/mmcblk0/size");
+        property_set("ro.frp.pst", "/dev/block/mmcblk0p12");
+    } else if (ro_hardware == "var-dart-mx6") {
+        property_set("ro.internel.storage_size", "/sys/block/mmcblk2/size");
+        property_set("ro.frp.pst", "/dev/block/mmcblk2p12");
+    }
+}
+
 void load_system_props() {
     load_properties_from_file(PROP_PATH_SYSTEM_BUILD, NULL);
     load_properties_from_file(PROP_PATH_VENDOR_BUILD, NULL);
     load_properties_from_file(PROP_PATH_FACTORY, "ro.*");
     load_recovery_id_prop();
+    variscite_system_props();
 }
 
 void start_property_service() {
-- 
2.7.4

